scripts/profile_get.py

```diff
@@ -0,0 +1,217 @@
+    print(sd_hash)
+    return sd_hash
+
+
+@defer.inlineCallbacks
+def benchmark_performance(uris: list, output_path: str) -> dict:
+    results = dict()
+    for uri in uris:
+        sd_hash = yield f2d(get_sd_hash_from_uri(uri))
+
+        start = time.time()
+        was_download_successful = yield download_blob_for_uri(sd_hash)
+        end = time.time()
+
+        if was_download_successful:
+            results[uri] = end - start
+        else:
+            results[uri] = "Could not download"
+
+        print(results[uri], uri, file=open(output_path, "a"))
+
+    return results
+
+
+def extract_uris(response):
+    uris = list()
+    for key in response:
+        for value in response[key]:
+            uris.append(value)
+
+    return uris
+
+
+@defer.inlineCallbacks
+def get_frontpage_uris():
+    kr = keyring.get_keyring()
+    c = kr.get_preferred_collection()
+    lbry_keyring = None
+    for col in c.get_all_items():
+        if col.get_label() == "LBRY/auth_token":
+            lbry_keyring = col
+    lbry_keyring = lbry_keyring.get_secret().decode("ascii")
+    response = yield treq.get("https://api.lbry.io/file/list_homepage?auth_token={}".format(lbry_keyring))
+    if response.code != 200:
+        log.error("API returned non 200 code!!")
+        reactor.callLater(0, reactor.stop)
+
+    body = yield response.json()
+    uris = extract_uris(body['data']['Uris'])
+    return uris
+
+
+@defer.inlineCallbacks
+def main(output_path):
+    global component_manager
+    yield component_manager.setup()
+    yield component_manager.get_component('dht')._join_deferred
+    uris = yield get_frontpage_uris()
+    # uris = [
+    #     "linux-kernel-patch-replaces-naughty#e4fc36ac921970b3d15138781d697841dfb745f7",
+    #     "linux-thursday-dec-1-2018-linux-y#062615ab5b974f5536b7e3d47731e7c74279ea23",
+    #     "what"
+    # ]
+    results = yield benchmark_performance(uris)
+    yield component_manager.stop()
+    shutil.rmtree(tempfile.tempdir)
+
+    max_len = len(max(uris, key=len))
+    _sum = num = 0
+    for result, value in results.items():
+        if not isinstance(value, str):
+            _sum += value
+            num += 1
+        print("{0:>{1:d}s}: {2}".format(result, max_len, value))
+
+    avg = _sum / num
+    print("Average for {0:d} downloaded files is: {1:f}".format(num, avg))
+    could_not_download = len(uris) - num
+    smiley = ":-)" if could_not_download == 0 else ":-("
+    print("{0:d} files could not be downloaded {1:s}".format(could_not_download, smiley))
+
+    reactor.callLater(0, reactor.stop)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--wallet_path", help="Enter None to use a temporary directory")
+    parser.add_argument("--output_path")
+    args = parser.parse_args()
+    wallet_path = args.wallet_path
+    output_path = args.output_path
+
+    tempfile.tempdir = tempfile.mkdtemp()
```

Comment:
> The module uses a global variable to store the name of the directory used for temporary files returned by gettempdir(). It can be set directly to override the selection process, **but this is discouraged. All functions in this module take a dir argument which can be used to specify the directory _and this is the recommended approach_.**

https://docs.python.org/3.7/library/tempfile.html#tempfile.gettempprefixb